clap_complete/src/dynamic/shells/mod.rs

```diff
@@ -16,6 +16,78 @@ use crate::dynamic::Completer as _;
 /// A subcommand definition to `flatten` into your CLI
 ///
 /// This provides a one-stop solution for integrating completions into your CLI
+///
+/// # Examples
+///
+/// The following example shows how to integrate completions into your CLI and generate completions.
+///
+/// 1. Build an application with derive API.
+/// 2. Call `clap_complete::dynamic::shells::CompleteCommand::augment_subcommands` to add the `complete` subcommand into the application.
+/// 3. Call `get_matches()`, or any of the other normal methods directly after.
+///
+/// For example:
+///
+/// ```no_run
+/// // src/main.rs
+/// use clap::{CommandFactory, FromArgMatches, Parser, Subcommand};
+/// use clap_complete::dynamic::shells::CompleteCommand;
+///
+/// #[derive(Parser, Debug)]
+/// #[clap(name = "dynamic", about = "A dynamic command line tool")]
+/// struct Cli {
+/// 	/// The subcommand to run complete
+/// 	#[command(subcommand)]
+/// 	complete: Option<CompleteCommand>,
+///     /// Input file path
+///     #[clap(short, long, value_hint = clap::ValueHint::FilePath)]
+///     input: Option<String>,
+///     /// Output format
+///     #[clap(short = 'F', long, value_parser = ["json", "yaml", "toml"])]
+///     format: Option<String>,
+/// }
+///
+/// fn main() {
+/// 	let cli = Cli::parse();
+///		if let Some(completions) = cli.complete {
+///			completions.complete(&mut Cli::command());
+///		}
+///
+/// 	// normal logic continues...
+/// }
```

Comment:
Is it just me or is the indentation off?

Reply:
![image](https://github.com/user-attachments/assets/c6b267cf-991a-47d0-83c3-1856040b34d9)
it's weird.

Reply:
It may be a problem with my vscode tab insert space configuration.

Reply:
Ah, yup, those are tabs